server/game/cards/17.1-R/AMissionInEssos.js

```diff
@@ -14,13 +14,13 @@ class AMissionInEssos extends DrawCard {
             handler: context => {
                 this.game.resolveGameAction(
                     GameActions.returnCardToHand(context => ({ card: context.target }))
-                        .then(preThenContext => ({
+                        .then((preThenContext, preThenEvent) => ({
                             gameAction: GameActions.search({
                                 title: 'Select a character',
                                 match: {
                                     type: 'character',
-                                    trait: preThenContext.target.getTraits(),
-                                    printedCostOrLower: preThenContext.target.getPrintedCost() - 1
+                                    printedCostOrLower: preThenEvent.cardStateWhenReturned.getPrintedCost() - 1,
+                                    condition: card => card.getTraits().some(trait => preThenEvent.cardStateWhenReturned.hasTrait(trait))
```

Comment:
I think with this one you could have just done `trait: preThenContext.event.cardStateWhenReturned.getTraits()`, since execution is delayed until the `then` clause is evaluated. But this also works. 🤷‍♀️